### PR TITLE
Feature/improve tests

### PR DESCRIPTION
**Pull request by jesjos**
1. the `Function` testing utility makes some hidden assumptions about which input is the expected and which is the function under test. Make this explicit.
2. Explicitly ignore errors in order to appease linters
3. Flip calls to `Function` in `quad_test` to reflect usage.

### CODE REVIEW

**Comment by jesjos** [comment]
Are there any maintainers available to look at PRs? E.g. @kigiri ?

File: lib/challenge/challenge.go [hunk 1]
```diff
@@ -77,48 +77,48 @@ func Monitor(fn interface{}, args []interface{}) (out Output) {
 	outC := make(chan string)
 	var buf strings.Builder
 	go func() {
-		io.Copy(&buf, r)
+		_, _ = io.Copy(&buf, r)
```
Comment (MSilva95) [inline lib/challenge/challenge.go:80]: just by adding the `_,_` doesn't take care of the error handling
But, we will take this into consideration and check for the best way to fix it.

File: tests/quad_test/main.go [hunk 2]
```diff
@@ -152,11 +152,11 @@ func main() {
 	// Tests all possibilities including 0 0, -x y, x -y
 	for i := 0; i < len(table); i += 2 {
 		if i != len(table)-1 {
-			challenge.Function("quadA", quadA, student.QuadA, table[i], table[i+1])
-			challenge.Function("quadB", quadB, student.QuadB, table[i], table[i+1])
-			challenge.Function("quadC", quadC, student.QuadC, table[i], table[i+1])
-			challenge.Function("quadD", quadD, student.QuadD, table[i], table[i+1])
-			challenge.Function("quadE", quadE, student.QuadE, table[i], table[i+1])
+			challenge.Function("quadA", student.QuadA, quadA, table[i], table[i+1])
```
Comment (MSilva95) [inline tests/quad_test/main.go:155]: this is already fixed here, you need to update your master
<https://github.com/01-edu/go-tests/pull/21/files>

**Review by MSilva95 — CHANGES_REQUESTED** [review]
If you would like to change only the names of the functions, please fix the pr accordingly.
Thank you for the feedback!

File: lib/challenge/challenge.go (hunk 1 above)
Comment (jesjos) [reply lib/challenge/challenge.go:80]: I'm not trying to "fix " error handling, I'm making it explicit that the original author chose not to do any error handling.

**Comment by mikysett** [comment]
@jesjos thanks a lot for your contribution, this PR will be merged [on this one](https://github.com/01-edu/go-tests/pull/164) since we made some little adjustments on variable names.

**Pull request closed without merging**